### DISCUSSION TITLE
Macintosh SE not booting

### DISCUSSION THREAD

**Discussion by CamWynd** [erichelgeson/BlueSCSI-v1#195] [Q&A] [answered]
Ok.
I've got an internal Bluescsi built and flashed. 
I found an SD card in my box of spares that works, and a boot image loaded on it.

My problem now is that when I connect it all up to my Mac SE and start it up, I end up with a ? disk on the the screen. I suspected maybe the scsi cable went bad, and replaced it; same result.

Has anyone else encountered this behaviour? Any pointers on further troubleshooting?

Thanks,
Cam

**Comment by erichelgeson** [comment]
I'd take a look at the https://github.com/erichelgeson/BlueSCSI/wiki/Troubleshooting guide - Use SD Card Formatter with full overwrite. Double check your solder work and all the components origination. Power ti via USB to rule out power issues, etc. All should be in that troubleshooting wiki page. Let us know what you find.

**Reply by CamWynd** [reply]
I went through all those steps, and even have the device writing a log file to the SD card:

BlueSCSI <-> SD - https://github.com/erichelgeson/BlueSCSI
VER: 1.1-20220917-USB
DEBUG:0 SDFAT_FILE_TYPE:3
SdFat version: 2.2.0
Sd Format: FAT32FAT16 - Consider formatting the SD Card with exFAT for improved performance.
SPI speed: 50Mhz
SdFat Max FileName Length: 64
Initialized SD Card - let's go!
Sd MID:27 OID:PH
Sd Name:2&DRP
Sd Date:4/2017
Sd Serial:26885421
 - HD60_512 6.0.8-500M.hda HDD / 524288000bytes / 512000KiB / 500MiB
ID:LUN0:
 0:----:
 1:----:
 2:----:
 3:----:
 4:----:
 5:----:
 6: 512:
Finished initialization of SCSI Devices - Entering main loop.

Which I'm assuming is the correct output (with the exception of the format type, I should change that over to exFAT).

I read that there might be some incompatibility with the SE, but also have read other people have no problem with the SE. Makes me wonder if there was a variation in the SCSI chip set in the Mac SE that might be a 'gotcha' for the BlueSCSI? Just in case anyone else has the same problem/suspicion, the SCSI chip in my SE is an AM5380PC made in 1986.

**Reply by xksteve** [reply]
Both mine have the same SCSI chip and work fine. Where I have had issues is with cheapo SD cards and a once working blueSCSI the stopped suddenly. This turned out to be a very slightly poor solder joint on the terminator. Both my busSCSIs work brilliantly (one internal, one external)
This is one of my logs in case it helps you (from my internal device - sanDisk 16gb in an adaptor via SD card extender)
Spec for clarity : Macintosh SE FDHD M5011 4Mb 1 x 1.44FD 1x BlueSCSI on internal Bus

Obvious point of note is I think you might need a shorter file name ??? 64 max think yours is over that but I keep loosing count (old age)

BlueSCSI <-> SD - https://github.com/erichelgeson/BlueSCSI
VER: 1.1-20220917-USB
DEBUG:0 SDFAT_FILE_TYPE:3
SdFat version: 2.2.0
Sd Format: exFAT
SPI speed: 50Mhz
SdFat Max FileName Length: 64
Initialized SD Card - let's go!
Sd MID:DF OID: 
Sd Name:SD
Sd Date:6/2012
Sd Serial:658
 - HD20_512 Legacy recovery 2GB HFS.hda HDD / 2147483648bytes / 2097152KiB / 2048MiB
Not an image: ._HD20_512 Legacy recovery 2GB HFS.hda
 - HD60_512 System 7.1.hda HDD / 2147483648bytes / 2097152KiB / 2048MiB
Not an image: 0_HD60_512 System 7.5.3.hda
 - HD40_512 Software install.hda HDD / 2147483648bytes / 2097152KiB / 2048MiB
Not an image: ._HD40_512 Software install.hda
Not an image: ._0_HD60_512 System 7.5.3.hda
Not an image: ._HD60_512 System 7.1.hda
ID:LUN0:
 0:----:
 1:----:
 2: 512:
 3:----:
 4: 512:
 5:----:
 6: 512:
Finished initialization of SCSI Devices - Entering main loop.

Hope you get this sorted as it is worth while 

All the best Steve

**Reply by xksteve** [reply]
Sorry mis read. its not the file name

**Reply by erichelgeson** [reply]
> I read that there might be some incompatibility with the SE

someone a long time ago said that, but we've actually never seen it. I bought a lot of SE's to test with and none failed. I do have a standing offer to exchange a SE motherboard with someone if they think this is the issue, but it is likely something else. 

Could you validate you've tested with a USB powered connection to rule out weak 5v on the SE? Could you try a different system 6 image, maybe one from the rascsi page on the macintosh garden?

**Reply by xksteve** [reply]
> Could you validate you've tested with a USB powered connection to rule out weak 5v on the SE? 

I forgot that I had to connect the power from the old HD to the blueSCSI. I even made up some leads especially for this purpose. the ext one powers fine from the SCSI plug on the back but my internal one would not boot - no LEDS on the Blue Pill - until I powered it (I have a question about that myself but thats for another thread).

I tried an old 8gb Samsung SD card this morning and even following the same procedure I used previously this does not even show in SCSI Probe as a device so I would defiantly try a different SD if the power idea does not prove fruitful.

Its also worth while checking your boot file works in an emulator and that you use one of the remade drives to boot, install and config to one of this projects downloadable blanks thus eliminating a HD driver issue.

**Reply by CamWynd** [reply]
Thanks for all the pointers..

- comparing the log files makes me believe my Bluescsi is working
- I'll try powering the board from the USB, in case the SE just has a weak power supply
- I've tried multiple system 6 and 7 images on a variety of micro SD cards and had the same results
- yep; they all boot in the Ballisk II emulator . . .
- built 2 more Bluescsi boards last night and I'll test with them today, but I'm suspecting it's a fault in the SE (either memory, SCSI controller, etc)
- I might pick up another Mac to test in

Thanks again!

**Comment by CamWynd** [comment, marked as answer]
Progress! 
I started checking continuity of all the connections (based on the 1.0 schematics). Everything was fine.
Then I got aggravated by such a simple project not working, and decided to check the resistor networks. THEY WERE WRONG. They were labeled 220 and 330 ohm, but measured waaaay below those values. I ordered new ones, replaced the bad parts, and it then it booted!
Now I get a 'Welcome to Macintosh' message and either a frozen desktop or a system error bomb message. I'm thinking either the image I'm trying to use is bad, or the SD card is.
